### PR TITLE
Return StencilInstantiation from IIRBuilder

### PR DESCRIPTION
**Pull request by havogt**
## Technical Description

Return `shared_ptr<StencilInstantiation>` from `IIRBuilder.build()`.

Reason:
- By design of the `IIRBuilder.build()`, the `StencilInstantiationContext` always contained exactly one `StencilInstantiation`. 
- The `shared_ptr<>` should be removed.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: dawn/src/dawn/Unittest/CompilerUtil.cpp [hunk 1]
```diff
@@ -69,20 +72,31 @@ void CompilerUtil::dumpNaive(std::ostream& os, dawn::codegen::stencilInstantiati
   os << ss.str();
 }
 
-void CompilerUtil::dumpCuda(std::ostream& os, dawn::codegen::stencilInstantiationContext& ctx) {
-  using CG = dawn::codegen::cuda::CudaCodeGen;
+dawn::codegen::stencilInstantiationContext
+si_to_context(std::shared_ptr<iir::StencilInstantiation> si) {
```
Comment (Stagno) [inline dawn/src/dawn/Unittest/CompilerUtil.cpp]:
```suggestion
siToContext(std::shared_ptr<iir::StencilInstantiation> si) {
```

File: dawn/src/dawn/Unittest/CompilerUtil.h [hunk 2]
```diff
@@ -15,16 +15,16 @@
...
-#include "dawn/CodeGen/Cuda/CudaCodeGen.h"
 #include "dawn/Compiler/DawnCompiler.h"
 #include "dawn/Compiler/Options.h"
+#include "dawn/IIR/StencilInstantiation.h"
 #include "dawn/Serialization/IIRSerializer.h"
 #include "dawn/Serialization/SIRSerializer.h"
 
 #include <fstream>
 #include <iostream>
+#include <string>
```
Comment (Stagno) [inline dawn/src/dawn/Unittest/CompilerUtil.h]: Not needed here, maybe in the cpp?

Comment (havogt) [reply dawn/src/dawn/Unittest/CompilerUtil.h]: used in load()

File: dawn/src/dawn/Unittest/CompilerUtil.cpp (hunk 1 above)
Comment (havogt) [reply dawn/src/dawn/Unittest/CompilerUtil.cpp]: thanks. fixed.

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: dawn/src/dawn/Unittest/CompilerUtil.h (hunk 2 above)
Comment (Stagno) [reply dawn/src/dawn/Unittest/CompilerUtil.h]: Right

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**